CRM/Utils/System/Joomla.php

```diff
@@ -38,10 +38,17 @@ public function __construct() {
    */
   public function createUser(&$params, $mail) {
     $baseDir = JPATH_SITE;
-    require_once $baseDir . '/components/com_users/models/registration.php';
-
     $userParams = JComponentHelper::getParams('com_users');
-    $model = new UsersModelRegistration();
+
+    if (version_compare(JVERSION, '4.0.0', 'ge')) {
+      $model = JFactory::getApplication()->bootComponent('com_users')->getMVCFactory()->createModel('Registration', 'Site');
+      $model->set('data', new \stdClass);
```

Comment:
@monishdeb can you change this to
```suggestion
      $model->set('data', new \stdClass());
```

Reply:
hmm didn't know that stylechecker didn't allow the without parenthesis new class declaration like [here](https://github.com/joomla/joomla-cms/blob/4.1-dev/components/com_users/src/Model/RegistrationModel.php#L293) 

Updating now.